modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/reducers/defaultReducer.es.js

```diff
@@ -69,6 +69,11 @@ export default (state, action) => {
 								}
 							}
 
+							try {
```

Comment:
@victorg1991, just so you know. I opted to attempt parsing at this moment of the response. Grid fields already provide values as JSON objects according to [this line](https://github.com/liferay/liferay-portal/blob/61df825f637edf4580aab5aa0c4c97c333a7cdcd/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/grid/GridDDMFormFieldTemplateContextContributor.java#L58-L67) and the reason why the React component was getting the value stringified is because of the `defaultReducer.es.js`. Before the components are rendered, the reducer will switch the value coming from the response to one that's inside the `localizedValue` property, which has all its values stringified. So, since this is a common problem to all fields, I did the parsing here.